client/src/context/Reservation/index.js

```diff
@@ -0,0 +1,15 @@
+import { createContext, useState } from 'react';
+
+export const ReservationContext = createContext();
+const ReservationProvider = ({ children }) => {
+  const [selctedSeats, setSelectedSeats] = useState([]);
```

Comment:
Did you consider using a useReducer? You could add different actions depending if you're adding or deleting.